File: backend/app/schemas/library.py
from __future__ import annotations

from pydantic import BaseModel

from app.schemas.project import ProjectRead, TagBrief


class FolderTreeNode(BaseModel):
    id: int
    name: str
    tags: list[TagBrief] = []
    children: list[FolderTreeNode]
    projects: list[ProjectRead]


FolderTreeNode.model_rebuild()


class LibraryTreeResponse(BaseModel):
    """语雀式侧边栏：顶层文件夹树 + 未归入文件夹的项目。"""

    folders: list[FolderTreeNode]
    orphan_projects: list[ProjectRead]

File: frontend/src/types/library.ts
import type { Project } from "@/types/project"

/** 与后端 `FolderTreeNode` / `LibraryTreeResponse` 对齐 */
export interface FolderTreeNode {
  id: number
  name: string
  tags?: FolderTagBrief[]
  children: FolderTreeNode[]
  projects: Project[]
}

export interface LibraryTreeResponse {
  folders: FolderTreeNode[]
  orphan_projects: Project[]
}

/** 与后端 TagBrief 对齐 */
export interface FolderTagBrief {
  id: number
  name: string
  category_id: number | null
}

export interface FolderRow {
  id: number
  parent_id: number | null
  name: string
  description: string | null
  sort_order: number
  tags: FolderTagBrief[]
  created_at: string
  updated_at: string
}
